packages/flame_studio/lib/src/widgets/flame_studio.dart

```diff
@@ -0,0 +1,67 @@
+import 'package:flame/game.dart';
+import 'package:flame_studio/src/widgets/scaffold.dart';
+import 'package:flame_studio/src/widgets/settings_provider.dart';
+import 'package:flame_studio/src/widgets/toolbar/pause_button.dart';
+import 'package:flame_studio/src/widgets/toolbar/start_button.dart';
+import 'package:flutter/widgets.dart';
+
+class FlameStudio extends StatefulWidget {
+  const FlameStudio(this.child, {super.key});
+
+  final Widget child;
+
+  @override
+  State<StatefulWidget> createState() => _FlameStudioState();
+
+  static final List<Widget> toolbarLeft = [];
+  static final List<Widget> toolbarMiddle = [
+    const StartButton(),
+    const PauseButton(),
+  ];
+  static final List<Widget> toolbarRight = [];
+}
+
+class _FlameStudioState extends State<FlameStudio> {
+  Game? _game;
+  Future<void>? _findGameWidgetFuture;
+  void _scheduleGameWidgetSearch() {
```

Comment:
I wonder if we shouldn't simply ask the game as an attribute for the `runFlameStudio` method, so we don't need to go through all of this code, I imagine that if the game widget is too deep in the tree, this could cause potential perfomance issues?

We could make the `Widget` parameter optional for the cases where the user only wants to pass the game directly, like:

```dart
runFlameStudio(MyGame());
```

```dart
final game = MyGame();
runFlameStudio(game, child: Container(child: GameWidget(game: game)));
```

Reply:
This could be problematic in some cases. For example, if the user is using `GameWidget.controlled()`, then the game instance is only known to the GameWidget's State.

In addition, it might be interesting to explore in the future the possibility of restarting the game from within the studio.